modules/xfem/src/base/XFEMGeometricCut2D.C

```diff
@@ -33,12 +33,17 @@ XFEMGeometricCut2D::cutElementByGeometry(const Elem* elem, std::vector<CutEdge>
 
   Real fraction = cutFraction(time);
 
-  unsigned int nsides = elem->n_nodes();
+  unsigned int n_sides = elem->n_sides();
 
-  for (unsigned int i = 0; i < nsides; ++i)
+  for (unsigned int i = 0; i < n_sides; ++i)
   {
-    const Node *node1 = elem->get_node(i);
-    const Node *node2 = elem->get_node(i<(nsides-1) ? i+1 : 0);
+    // This returns the lowest-order type of side, which should always
+    // be an EDGE2 here because this class is for 2D only.
+    UniquePtr<Elem> curr_side = elem->side(i);
+    if (curr_side->type() != EDGE2)
+      mooseError("Invalid element type in cutElementByGeometry");
```

Comment:
Yes:
http://libmesh.github.io/doxygen/namespacelibMesh_1_1Utility.html#aa99a42a5efab878469e7f58e5c033335

On Mon, Feb 29, 2016 at 11:46 AM Ben Spencer <notifications@github.com>
wrote:

> In modules/xfem/src/base/XFEMGeometricCut2D.C
> <https://github.com/idaholab/moose/pull/6451#discussion_r54456421>:
>
> >    {
> > -    const Node *node1 = elem->get_node(i);
> > -    const Node *node2 = elem->get_node(i<(nsides-1) ? i+1 : 0);
> > +    // This returns the lowest-order type of side, which should always
> > +    // be an EDGE2 here because this class is for 2D only.
> > +    UniquePtr<Elem> curr_side = elem->side(i);
> > +    if (curr_side->type() != EDGE2)
> > +      mooseError("Invalid element type in cutElementByGeometry");
>
> I modified the code to print out the type, but it's just an enum, which
> isn't terribly helpful either. Is there any way to convert the enums to
> strings to print out the type?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/6451/files#r54456421>.
>